python/rikai/types/vision.py

```diff
@@ -40,10 +49,61 @@ class Image(ToNumpy, Asset, Displayable):
 
     __UDT__ = ImageType()
 
-    def __init__(self, uri: str):
+    def __init__(self, uri: Union[str, Path]):
         super().__init__(uri)
         self._cached_data = None
 
+    @classmethod
+    def from_array(
+        cls,
+        array: np.ndarray,
+        uri: Union[str, Path],
+        mode: str = None,
+    ) -> Image:
+        """Create an image in memory from numpy array.
+
+        Parameters
+        ----------
+        array : np.ndarray
+            Array data
+        uri : str or Path
+            The external URI to store the data.
+        mode : str, optional
+            The mode which PIL used to create image. See supported
+            `modes on PIL document <https://pillow.readthedocs.io/en/stable/handbook/concepts.html#concept-modes>`_.
+
+        See Also
+        --------
+        :py:class:`PIL.Image.fromarray`
+        :py:func:`~rikai.spark.functions.vision.numpy_to_image`
+
+        """  # noqa: E501
+
+        assert array is not None
+        img = PILImage.fromarray(array, mode=mode)
+        return cls.from_pil(img, uri)
+
+    @staticmethod
+    def from_pil(img: PILImage, uri: Union[str, Path]) -> Image:
+        """Create an image in memory from a :py:class:`PIL.Image`.
+
+        Parameters
+        ----------
+        img : :py:class:`PIL.Image`
+            An PIL Image instance
+        uri : str or Path
+            The URI to store the image externally.
+        """
+        parsed = urlparse(normalize_uri(uri))
+        if parsed.scheme == "file":
+            img.save(uri)
```

Comment:
I *think* this should also work directly for dbfs

Reply:
you meant `PIL` can directly save to `dbfs`?

Reply:
We could standardize to get a `write fobj / stream` from `uri`, and make them works with these third party libs that works with `fobj`.